### PR TITLE
[cloud-azure] Split azure plugin in 3 plugins

### PR DESCRIPTION
**Pull request by dadoonet**
Until now we had a cloud-azure plugin which is providing 3 distinct features:
- discovery on Azure
- snapshot/restore on Aure
- SMB store

This commit splits the plugin by feature so people can use either one or the other or both features.

Doc is updated accordingly.

### CODE REVIEW

File: docs/plugins/store-smb.asciidoc [hunk 1]
```diff
@@ -0,0 +1,80 @@
+[[store-smb]]
+=== Store SMB Plugin
+
+The Store SMB plugin proposes a work around for a bug in Windows SMB and Java on windows.
```
Comment (rjernst) [inline docs/plugins/store-smb.asciidoc]: "proposes a work around for" -> "works around"

File: docs/plugins/store.asciidoc [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+Store plugins offer alternatives to default Lucene stores.
+
+[float]
+=== Core store plugins
+
+The core store plugins are:
+
+<<store-smb,Store SMB>>::
+
+The Store SMB plugin proposes a work around for a bug in Windows SMB and Java on windows.
```
Comment (rjernst) [inline docs/plugins/store.asciidoc]: ditto on wording suggested above

File: docs/reference/modules/discovery/azure.asciidoc [hunk 3]
```diff
@@ -1,6 +1,4 @@
 [[modules-discovery-azure]]
 === Azure Discovery
 
-Azure discovery allows to use the Azure APIs to perform automatic discovery (similar to multicast).
-Please check the https://github.com/elasticsearch/elasticsearch-cloud-azure[plugin website]
-to find the full documentation.
+Azure discovery is available as a plugin. See {plugins}/discovery-azure.html[discovery-azure] for more information.
```
Comment (rjernst) [inline docs/reference/modules/discovery/azure.asciidoc]: why the removal of the old single line description? I think that is better than just saying "there is a plugin"

File: plugins/repository-azure/licenses/azure-LICENSE.txt [hunk 4]
```diff
@@ -0,0 +1,1565 @@
+
+
+
+<!DOCTYPE html>
```
Comment (rjernst) [inline plugins/repository-azure/licenses/azure-LICENSE.txt]: I see this hasn't changed but...is this really the license? Looks like garbage html, not a txt license as the file suffix suggests...

File: plugins/store-smb/src/main/java/org/elasticsearch/plugin/store/smb/SMBStorePlugin.java [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+ */
+
+package org.elasticsearch.plugin.store.smb;
+
+import org.elasticsearch.index.store.IndexStoreModule;
+import org.elasticsearch.index.store.smbmmapfs.SmbMmapFsIndexStore;
+import org.elasticsearch.index.store.smbsimplefs.SmbSimpleFsIndexStore;
+import org.elasticsearch.plugins.Plugin;
+
+/**
```
Comment (rjernst) [inline plugins/store-smb/src/main/java/org/elasticsearch/plugin/store/smb/SMBStorePlugin.java]: Remove empty javadoc? Or actually fill it in...

**Comment by rjernst** [comment]
LGTM. I left a few minor non-code comments.

File: docs/plugins/store-smb.asciidoc (hunk 1 above)
Comment (dadoonet) [inline docs/plugins/store-smb.asciidoc]: agreed. Thanks

File: docs/plugins/store.asciidoc (hunk 2 above)
Comment (dadoonet) [inline docs/plugins/store.asciidoc]: agreed. Thanks

File: docs/reference/modules/discovery/azure.asciidoc (hunk 3 above)
Comment (dadoonet) [inline docs/reference/modules/discovery/azure.asciidoc]: Well I tried to have the same structure as for ec2: https://github.com/elastic/elasticsearch/blob/master/docs/reference/modules/discovery/ec2.asciidoc

BTW with your comment I found a link error in https://github.com/elastic/elasticsearch/blob/master/docs/reference/modules/discovery/gce.asciidoc. I think I'll open a doc PR to fix that and ec2 doc.

But for now, I will apply your comment to azure doc.

**Pull request merged**